pkg/services/ngalert/notifier/receiver_groups.go

```diff
@@ -0,0 +1,227 @@
+package notifier
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"slices"
+
+	"github.com/grafana/alerting/notify"
+
+	"github.com/prometheus/alertmanager/config"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+var (
+	// TODO: convert to errutil
+	ErrPermissionDenied = errors.New("permission denied")
+)
+
+type ReceiverGroupService struct {
+	ac                 accesscontrol.AccessControl
+	provisioningStore  provisoningStore
+	versionedConfStore *LockingConfigStore
+	encryptionService  secrets.Service
+	xact               TransactionManager
+	log                log.Logger
+}
+
+type provisoningStore interface {
+	GetProvenances(ctx context.Context, org int64, resourceType string) (map[string]models.Provenance, error)
+}
+
+type TransactionManager interface {
+	InTransaction(ctx context.Context, work func(ctx context.Context) error) error
+}
+
+func NewReceiverGroupService(
+	ac accesscontrol.AccessControl,
+	configStore configStore,
+	provisioningStore provisoningStore,
```

Comment:
We can't import from provisioning here because we introduce an import cycle. We could move the store and its public interface to the `store` package, but that can also be revisited later as this service is built out.